en/02_Developer_Guides/14_Files/05_File_Manipulation.md

```diff
@@ -0,0 +1,419 @@
+| `File::getMimeType`      | Get the mime type of this file                             |
+| `File::getMetaData`      | Gets other metadata from the file as an array              |
+| `File::getFileType`      | Return the type of file for the given extension            |
+
+### Additional file types
+
+Silverstripe CMS has a pre-defined list of common file types. `File::getFileType` will return "unknown" for files outside that list.
+
+You can add your own file extensions and their description with the following configuration:
+
+```yml
+SilverStripe\Assets\File:
+  file_types:
+    ai: 'Adobe Illustrator'
+    psd: 'Adobe Photoshop File'
+```
+
+## Renaming and moving files
+
+In order to move or rename a file you can simply update the `Name` property, or assign the `ParentID` to a new
+folder. Please note that these modifications are made simply on the draft stage, and will not be copied
+to live until a publish is made via the CMS (either on this object, or cascading from a parent).
+
+When files are renamed using the ORM, all file variants are automatically renamed at the same time.
+
+```php
+use SilverStripe\Assets\File;
+
+$file = File::get()->filter('Name', 'oldname.jpg')->first();
+if ($file) {
+    // The below will move 'oldname.jpg' and 'oldname__variant.jpg'
+    // to 'newname.jpg' and 'newname__variant.jpg' respectively
+    $file->Name = 'newname.jpg';
+    $file->write();
+}
+```
+
+Note that you can cause the file to be moved immediately by [setting the Versioned reading mode](api:SilverStripe\Versioned\Versioned::set_reading_mode()) to draft temporarily.
+
+```php
+use SilverStripe\Assets\File;
+use SilverStripe\Versioned\Versioned;
+
+$file = File::get()->filter('Name', 'oldname.jpg')->first();
+if ($file) {
+    // The below will immediately move 'oldname.jpg' and 'oldname__variant.jpg'
+    // to 'newname.jpg' and 'newname__variant.jpg' respectively
+    $file->Name = 'newname.jpg';
+    Versioned::withVersionedMode(function () use ($file) {
+        Versioned::set_reading_mode('Stage.' . Versioned::DRAFT);
+        $file->write();
+        $file->publishSingle();
+    });
+}
+```
+
+## Convert a file to a different format
+
+You can use the [`manipulateExtension()`](api:SilverStripe\Assets\ImageManipulation::manipulateExtension()) method on any `File` or `DBFile` object to create a variant with a different file extension than the original.
+
+This can be very useful if you want to convert a file to a different format for the user to download or view, while leaving the original file intact. Some examples of when you might want this are:
+
+- Generating thumbnails for videos, documents, etc
+- Converting images to `.webp` for faster page load times
+- Converting documents to `.pdf` so downloaded documents are more portable
+
+### Converting between image formats
+
+Converting between image formats is the easiest example, because we can let [Intervention Image](https://image.intervention.io/v2) do the heavy lifting for us.
```

Comment:
Why are we documenting how users can create an extension in their project code to do the conversion?

If we just an existing library to do the conversion shouldn't we just put this code straight into core?

Reply:
a) it demonstrates the concept, and
b) putting this code straight into core is not in scope. There's a follow-up card to do it, but since it hasn't been done yet I'm adding documentation. Could be we add that in tomorrow, but more likely it'll be a while before someone gets to it.

Reply:
We'll just remove this section docs once we do that card though

Doesn't seem like much point peer reviewing something that's just going to get removed

Reply:
I don't think we will remove this section because
> a) it demonstrates the concept

But even if we do, this documentation should be here until that point, because until then we still need to explain how to do this.